egs2/TEMPLATE/asr1/pyscripts/utils/extract_xvectors.py

```diff
@@ -42,6 +44,82 @@ def get_parser():
     return parser
 
 
+class XVExtractor:
+    def __init__(self, args, device):
+        self.toolkit = args.toolkit
+        self.device = device
+        from speechbrain.dataio.preprocess import AudioNormalizer
+
+        self.audio_norm = AudioNormalizer()
+        if self.toolkit == "speechbrain":
+            from speechbrain.pretrained import EncoderClassifier
+
+            self.model = EncoderClassifier.from_hparams(
+                source=args.pretrained_model, run_opts={"device": device}
+            )
+        elif self.toolkit == "rawnet":
+            if not os.path.exists("RawNet"):
+                git.Repo.clone_from("https://github.com/Jungjee/RawNet", "RawNet")
+            repo = git.Repo("RawNet")
+            repo.git.submodule("update", "--init", "--recursive")
+            sys.path.append("RawNet/python/RawNet3/models")
+            sys.path.append("RawNet/python/RawNet3")
+            subprocess.check_call(
+                [sys.executable, "-m", "pip", "install", "asteroid-filterbanks==0.4.0"]
+            )
```

Comment:
Could you remove this part and make `tools/installers/install_rawnet`?
Also, if you want to add python path, you can use `tools/extra_path.sh`

Reply:
Thanks a lot for the suggestion. I have refactored the installation.